### PR TITLE
Add summary-only flag to github_team data source

### LINKED ISSUE #1377: [FEAT]: Add data.github_team summary_only flag
### Describe the need

The data source "github_team" should support summary_only flag similar to data source "organization_teams" 
https://registry.terraform.io/providers/integrations/github/latest/docs/data-sources/organization_teams#summary_only

The github_team source downloads all the members and repos associated with the team, which makes it take O(minutes) whereas getting github_organization_teams takes O(seconds)

Here is where github_team source downloads all the members and repos:
https://github.com/integrations/terraform-provider-github/blob/bb0f993d8510d1673ac5bc47dd8b3455d5f28312/github/data_source_github_team.go#L83

Example:
```
# Define a repo, give a bunch of teams access by name (teams defined by IDP)

resource "github_repository" "default" {
# ...
}

locals = {
  # team slugs
  teams = ["team-one", "team-two", ...]
}

data "github_team" "the_team" {
  for_each = local.teams
  slug     = each.key
}

resource "github_team_repository" "team_repo" {
  for_each = local.teams

  team_id    = data.github_team.team[each.key].id

  repository = github_repository.default.name
  permission = ...
}
```

If you are struggling with this, we are working around by doing
```
# Define a repo, give a bunch of teams access by name (teams defined by IDP)

resource "github_repository" "default" {
# ...
}

locals = {
  # team slugs
  teams = ["team-one", "team-two", ...]
}

# Rather than download the teams needed, download all teams in summary_only view
data "github_organization_teams" "all" {
  summary_only = "true"
}

resource "github_team_repository" "team_repo" {
  for_each = {
    for team in data.github_organization_teams.all.teams
  : team.slug => team.id if contains(keys(local.teams), team.slug) }

  team_id    = data.github_team.team[each.key].id

  repository = github_repository.default.name
  permission = ...
}
```

### SDK Version

5.8.0 (latest)

### API Version

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by david-bain**
Add a flag input to optionally skip returning the repositories and members on calling the github_team data source.

This improves performance of the data source call when only summary information is needed.
Especially important on orgs with large numbers of teams/members.

* Add summary_only optional flag input to skip returning members and repositories
* Add 'results_per_page' optional int to reduce the results per page returned.
* Update docs to reflect changes

Resolves #1377

----

### Before the change?

Each call to github_teams would also make a call to get the team members and repositories. When there are a large number of teams that need querying, this could add additional time when only the team details were needed.

### After the change?

This improves performance of the data source when only summary information is needed by skipping API calls to get repositories and members. Especially important on orgs with large numbers of teams/members that require calling this data source multiple times.

### Other information

Basically a replication of #1176 but for single teams.

----

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No
- [ ] 
No, additional inputs are optional with defaults if not set resulting in no change to behaviour.

If `Yes`, what's the impact:  

* N/A

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

File: website/docs/d/team.html.markdown [hunk 1]
```diff
@@ -19,17 +19,18 @@ data "github_team" "example" {
...
- * `repositories` - List of team repositories (list of repo names)
-
+* `id` - the ID of the team.
+* `node_id` - the Node ID of the team.
+* `name` - the team's full name.
+* `description` - the team's description.
+* `privacy` - the team's privacy type.
+* `permission` - the team's permission level.
+* `members` - List of team members (list of GitHub usernames)
+* `repositories` - List of team repositories (list of repo names)
```
Comment (alexlo03) [inline website/docs/d/team.html.markdown:36]: Suggestion: add ""Not returned if `summary_only = true`"" to each

**Review by alexlo03 — COMMENTED** [review]
LGTM (FWIW)

File: website/docs/d/team.html.markdown (hunk 1 above)
Comment (david-bain) [reply website/docs/d/team.html.markdown]: added, thanks :)

**Review by kfcampbell — APPROVED** [review]
Confirming that integration tests are passing for me locally. Thanks for contributing again, @david-bain!

**Pull request merged**